src/i965_encoder.c

```diff
@@ -1453,6 +1468,16 @@ intel_enc_hw_context_init(VADriverContextP ctx,
             encoder_context->quality_range = ENCODER_QUALITY_RANGE_AVC;
         } else
             encoder_context->quality_range = ENCODER_QUALITY_RANGE;
+
+        if (obj_config->entrypoint == VAEntrypointFEI) {
+            encoder_context->fei_enabled = 1;
```

Comment:
If the code reaches intel_enc_hw_context_init() code block, the HAS_FEI_H264_ENCODING(i965) is always TRUE, because otherwise, config creation could have been already failed well ahead in vaCreateConfig.

But, one optimization is possible, which is to avoid the IS_SKL() check coming inside the (IS_SKL || IS_BXT) code block since fei_enabled == TRUE happens only for SKL.